### PR TITLE
Tests against a local piwik instance - fix #24

### LINKED ISSUE #24: Decent test coverage
Especially on edge cases:
- [ ] Piwik unresponsive
- [ ] Bad API key

### PR DESCRIPTION
**Pull request by abulte**
Some basic testing of views and downloads count and aggregation.

Next steps: #22 and #23.

### CODE REVIEW

File: tests/client.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+from udata_piwik.client import track
+
+from .conftest import PiwikSettings
+
+BASE_URL = 'http://{0.PIWIK_URL}'.format(PiwikSettings)
+RESET_URL = '{0}/reset.php'.format(BASE_URL)
+
+
+@retry(stop_max_attempt_number=20, wait_fixed=500)
```
Comment (noirbizarre) [inline tests/client.py]: Thanks for this,I was looking for something like that :ok_hand:

File: tests/client.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
```
Comment (noirbizarre) [inline tests/client.py]: To avoid relative import, I think `client` can become a class into `conftest.py` available as a fixture

**Comment by abulte** [comment]
@noirbizarre I put the `docker-compose.yml` file in https://github.com/opendatateam/udata-piwik-docker. Which when I did the PR seemed logical since the image was not available on docker hub. Now that it is, I certainly should put it here...

File: tests/client.py (hunk 1 above)
Comment (abulte) [reply tests/client.py]: ❤️

File: tests/client.py (hunk 2 above)
Comment (abulte) [reply tests/client.py]: I see your point. I can live with that for now though ;-)
